drivers/arm/pl011/aarch64/pl011_console.S

```diff
@@ -13,15 +13,72 @@
  */
 #include "../../../console/aarch64/console.S"
 
+	.globl	console_pl011_init
+	.globl	console_pl011_putc
+	.globl	console_pl011_getc
+	.globl	console_pl011_flush
 
-	.globl	console_core_init
-	.globl	console_core_putc
-	.globl	console_core_getc
-	.globl	console_core_flush
+#if MULTI_CONSOLE_API
+	.globl console_pl011_register
 
+	/* -----------------------------------------------
+	 * int console_pl011_register(console_pl011_t *console,
+		uintptr_t base, uint32_t clk, uint32_t baud)
+	 * Function to initialize and register a new PL011
+	 * console. Storage passed in for the console struct
+	 * *must* be persistent (i.e. not from the stack).
+	 * In: x0 - pointer to empty console_pl011_t struct
+	 *     x1 - UART register base address
+	 *     w2 - UART clock in Hz
+	 *     w3 - Baud rate
+	 * Out: return 1 on success, 0 on error
+	 * Clobber list : x0, x1, x2, x6, x7, x14
+	 * -----------------------------------------------
+	 */
+func console_pl011_register
```

Comment:
Again, I suspect this will collapse into a very simple function if implemented in C.